app/uk/gov/hmrc/excisemovementcontrolsystemapi/scheduling/PollingNewMessagesWithWorkItemJob.scala

```diff
@@ -136,15 +136,19 @@ class PollingNewMessagesWithWorkItemJob @Inject()
     } yield success
   }
 
-  private def process(messages: Seq[IEMessage], exciseNumber: String): Future[Boolean] = {
+  private def process(messages: Seq[IEMessage], exciseNumber: String): Future[Boolean] =
```

Comment:
what about doing something like this as may be more readable:

`
private def process1(messages: Seq[IEMessage], exciseNumber: String): Future[Boolean] = {
    messages.traverse(message => {
      println(s"1)== messageType: ${message.messageType}")
      for {
        isMessageSaved: Boolean <- saveToDbAndSendNotification(message, exciseNumber)
        _ =  if(isMessageSaved) auditService.auditMessage(message).value
        _ = if(!isMessageSaved) auditService.auditMessage(message, "Failed to process").value
      } yield isMessageSaved
    })
      .map(result => !result.contains(false) && result.nonEmpty)
  }

`

so we eliminate the nested if statement.  Also this show an possible problem with the test as the auditService.auditMessage(any, any)(any) mock expectation has never bees net and this can cause a nullpointerexception if we use the result. The are also some tests case missing for the audit. For example in PollingNewMessagesWithWorkItemJobSpec in the "any individual message is not saved to the database" test we can test how may time the oudit is call for successful and unsuccsefful save to the DB something like below:

`
"any individual message is not saved to the database" in {
        addOneItemToMockQueue()
        when(newMessageService.getNewMessages(any)(any))
          .thenReturn(Future.successful(Some((newMessageResponse, 5))))

        when(auditService.auditMessage(any, any)(any)).thenReturn(EitherT.fromEither(Right(())))
        when(newMessageParserService.extractMessages(any)).thenReturn(Seq(message, message, message, message, message))
        when(movementService.updateMovement(any, any))
          .thenReturn(
            Future.successful(Seq(Movement(Some("boxId1"), "lrn1", "consignor", Some("consignee")))),
            Future.successful(Seq(Movement(Some("boxId2"), "lrn2", "consignor", Some("consignee")))),
            Future.successful(Seq.empty),
            Future.successful(Seq(Movement(Some("boxId3"), "lrn3", "consignor", Some("consignee")))),
            Future.successful(Seq(Movement(Some("boxId4"), "lrn4", "consignor", Some("consignee")))),
          )

        await(job.executeInMutex)

        verify(newMessageService, never()).acknowledgeMessage(any)(any)
        verify(notificationService, times(4)).sendNotification(any, any, any)(any)

        withClue("should audit successful message") {
          verify(auditService, times(4)).auditMessage(eqTo(message))(any)
        }

        withClue("shoulld audit failing message") {
          verify(auditService, times(1)).auditMessage(eqTo(message), eqTo("Failed to process"))(any)
        }
      }
`

Reply:
Updated